src/lib/db/feature-strategy-store.ts

```diff
@@ -214,27 +222,53 @@ class FeatureStrategiesStore implements IFeatureStrategiesStore {
 
     async getFeatureToggleWithEnvs(
         featureName: string,
+        userId?: number,
         archived: boolean = false,
     ): Promise<FeatureToggleWithEnvironment> {
-        return this.loadFeatureToggleWithEnvs(featureName, archived, false);
+        return this.loadFeatureToggleWithEnvs({
+            featureName,
+            archived,
+            withEnvironmentVariants: false,
+            userId,
+        });
     }
 
     async getFeatureToggleWithVariantEnvs(
         featureName: string,
+        userId?: number,
         archived: boolean = false,
     ): Promise<FeatureToggleWithEnvironment> {
-        return this.loadFeatureToggleWithEnvs(featureName, archived, true);
+        return this.loadFeatureToggleWithEnvs({
+            featureName,
+            archived,
+            withEnvironmentVariants: true,
+            userId,
+        });
     }
 
-    async loadFeatureToggleWithEnvs(
-        featureName: string,
-        archived: boolean,
-        withEnvironmentVariants: boolean,
-    ): Promise<FeatureToggleWithEnvironment> {
+    async loadFeatureToggleWithEnvs({
+        featureName,
+        archived,
+        withEnvironmentVariants,
+        userId,
+    }: ILoadFeatureToggleWithEnvsParams): Promise<FeatureToggleWithEnvironment> {
         const stopTimer = this.timer('getFeatureAdmin');
-        const rows = await this.db('features_view')
+        let query = this.db('features_view')
             .where('name', featureName)
             .modify(FeatureToggleStore.filterByArchived, archived);
+
+        let selectColumns = ['features_view.*'];
+        if (userId && this.flagResolver.isEnabled('favorites')) {
+            query = query.leftJoin(`favorite_features as ff`, function () {
+                this.on('ff.feature', 'features_view.name').andOnVal(
```

Comment:
I'd not mix abbreviations with full names. If we have features_view I'd use favorite_features instead of ff